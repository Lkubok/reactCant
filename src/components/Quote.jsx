import React, { Component } from "react";
import { connect } from "react-redux";
import { updateQuoteTwo } from "../actions";
import MainQuote from "./QuoteComponents/MainQuote";

export class Quote extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.updateQuoteTwo();
  }

  render() {
    return (
      <div className="container d-flex flex-column justify-content-center">
        <MainQuote />
        <button className="btn btn-light m-3" onClick={this.handleChange}>
          Fetch new Quote 2
        </button>
      </div>
    );
  }
}

/* const mapStateToProps = (state, ownProps) => ({
  quote: state.quoteReducer.mainQuote,
  author: state.quoteReducer.author
}); */

const mapDispatchToProps = {
  updateQuoteTwo
};

export default connect(
  null,
  mapDispatchToProps
)(Quote);
