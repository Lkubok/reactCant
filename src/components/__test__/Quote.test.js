import React from "react";
import { shallow } from "enzyme";
import { Quote } from "../Quote";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Quote debug />);

    expect(component).toMatchSnapshot();
  });
});
