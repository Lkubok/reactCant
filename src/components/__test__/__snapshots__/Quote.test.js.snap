// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="container d-flex flex-column justify-content-center"
>
  <Connect(MainQuote) />
  <button
    className="btn btn-light m-3"
    onClick={[Function]}
  >
    Fetch new Quote 2
  </button>
</div>
`;
